srml/aura/src/lib.rs

```diff
@@ -176,7 +196,93 @@ decl_storage! {
 }
 
 decl_module! {
-	pub struct Module<T: Trait> for enum Call where origin: T::Origin { }
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		/// Report equivocation in block production.
+		fn report_equivocation(_origin, _equivocation_proof: Vec<u8>) {
```

Comment:
Why can't the `equivocation_proof` be strongly typed as `AuraEquivocationProof`?

Do I understand right that this is invoked as a no-op after `ValidateUnsigned` checks the call and this is the place where `slashing::slash` would be invoked?

Reply:
Yes, exactly.